File: 14315-2754-122-best-time-to-buy-and-sell-stock-ii/14315-2754-122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int result = 0;
        
        for(int i = 0; i < prices.length - 1; i++) {
            // 주식의 가격이 오른다면 항상 이익을 취한다.
            if(prices[i] < prices[i+1]) {
                result += (prices[i+1] - prices[i]);
            }
        }
        
        return result;
    }
}

File: 14315-2754-122-best-time-to-buy-and-sell-stock-ii/14315-2754-122-best-time-to-buy-and-sell-stock-ii.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var result = 0
        
        for (i in 0 until prices.size - 1) {
            // 주식의 가격이 오른다면 항상 이익을 취한다.
            if(prices[i] < prices[i+1]) {
                result += (prices[i+1] - prices[i])
            }
        }

        return result
    }
}